libsbom/spdx.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import datetime
+import uuid
+
+from spdx_tools.spdx.writer.json     import json_writer
+from spdx_tools.spdx.writer.tagvalue import tagvalue_writer
+from spdx_tools.spdx.writer.xml      import xml_writer
+from spdx_tools.spdx.writer.yaml     import yaml_writer
+from spdx_tools.spdx.model import (
+    Actor,
+    ActorType,
+    Annotation,
+    AnnotationType,
+    Checksum,
+    ChecksumAlgorithm,
+    CreationInfo,
+    Document,
+    ExternalPackageRef,
+    ExternalPackageRefCategory,
+    Package,
+    PackagePurpose,
+    Relationship,
+    RelationshipType
+)
+
+from cas_wrapper import CasWrapper
+from version import __version__
+
+ALMAOS_VENDOR = 'AlmaLinux OS Foundation'
+ALMAOS_EMAIL = 'cloud-infra@almalinux.org'
+ALMAOS_SBOMLICENSE = 'CC0-1.0'                  # FIXME: Determine license for AlmaLinux SBOMs
+ALMAOS_NAMESPACE = 'https://security.almalinux.org/spdx'
+
+TOOLS = [
+    {
+        "vendor": ALMAOS_VENDOR,
+        "name": "AlmaLinux Build System",
+        "version": "0.1"
+    }, {
+        "vendor": ALMAOS_VENDOR,
+        "name": "alma-sbom",
+        "version": __version__
+    }, {
+        "vendor": ALMAOS_VENDOR,
+        "name": "spdx-tools",
+        "version": "0.0" # FIXME: Need correct version info for spdx-tools
```

Comment:
The problem is that the required version has not been released yet, so all I can say is that the version will be newer than v0.7.1.

Reply:
I just added a commit that freezes the required version, since it has recently been announced. But: it has not been released to PyPI yet, so it will probably cause some issues in the meanwhile.